packages/graphql/src/translate/create-auth-and-params.test.ts

```diff
@@ -94,6 +94,92 @@ describe("createAuthAndParams", () => {
                 this_auth_allow0_id: sub,
             });
         });
+
+        test("should combine roles with where across rules", () => {
+            const idField = {
+                fieldName: "id",
+                typeMeta: {
+                    name: "ID",
+                    array: false,
+                    required: false,
+                    pretty: "String",
+                    input: {
+                        where: {
+                            type: "String",
+                            pretty: "String",
+                        },
+                        create: {
+                            type: "String",
+                            pretty: "String",
+                        },
+                        update: {
+                            type: "String",
+                            pretty: "String",
+                        },
+                    },
+                },
+                otherDirectives: [],
+                arguments: [],
+            };
+
+            const node = new NodeBuilder({
+                name: "Movie",
+                relationFields: [],
+                cypherFields: [],
+                enumFields: [],
+                scalarFields: [],
+                primitiveFields: [idField],
+                temporalFields: [],
+                interfaceFields: [],
+                objectFields: [],
+                pointFields: [],
+                auth: {
+                    rules: [
+                        {
+                            operations: ["READ"],
+                            roles: ["admin"],
+                        },
+                        {
+                            operations: ["READ"],
+                            roles: ["member"],
+                            where: { id: "$jwt.sub" },
+                        },
+                    ],
+                    type: "JWT",
+                },
+            }).instance();
+
+            // @ts-ignore
+            const neoSchema: Neo4jGraphQL = {
+                nodes: [node],
+            };
+
+            const sub = generate({
+                charset: "alphabetic",
+            });
+
+            // @ts-ignore
+            const context: Context = { neoSchema };
+            context.jwt = {
+                sub,
+            };
```

Comment:
Here, we should try to avoid `@ts-ignore`, for example using `ContextBuilder`

```suggestion
            const context = new ContextBuilder({
                jwt: {
                    sub,
                },
            }).instance();
```